forte/data/ontology/top.py

```diff
@@ -1078,13 +1111,18 @@ class Box(Region):
         width: the width of the box, the unit is one image array entry.
     """
 
+    _cy: int
+    _cx: int
+    _height: int
+    _width: int
+
     def __init__(
         self,
         pack: PackType,
-        cy: int,
-        cx: int,
-        height: int,
-        width: int,
+        cy: int = 0,
+        cx: int = 0,
+        height: int = 1,
+        width: int = 1,
```

Comment:
So the reason we had defaults was that during converting `DataStore` entries to objects (and when the entry is not present in `DataIndex`, we reach [here](https://github.com/Pushkar-Bhuse/forte/blob/76e16664002c50d91d7214eb6a6f553603a93f4f/forte/data/entry_converter.py#L212). Since we only pass the pack, it throws an error. That was the reason we had added default values earlier. But I'll discuss a better fix for this with @hepengfe.

Reply:
I will leave it to you to study the resolution. But please note that the principle should be "API drives the implementation", it can never be "the implementation drives the API".

On this issue, @mylibrar can probably help you more.

Reply:
Yes so I discussed this with @mylibrar. The solution we came up with was - When converting a datastore to object, the default values are just a placeholder and they are immediately replaced by values stored in the DataStore entry. Also, this case is very specific to `Box` and `BoundingBox` type entries. Thus, we can have separate `elif` branches before [this](https://github.com/Pushkar-Bhuse/forte/blob/76e16664002c50d91d7214eb6a6f553603a93f4f/forte/data/entry_converter.py#L208) to just pass dummy values to `Box` and `BoundingBox` (instead of setting default parameter values). These dummy values will then be overwritten by the values stored in the DataStore entry.

Reply:
The solution is another issue, but the interface is important, we cannot fit the interface to implementation. So if the default values don't make sense, we can't have them.